src/RecipeBot.Persistence/RecipeRepository.cs

```diff
@@ -86,14 +86,35 @@ public async Task<RecipeEntryData> DeleteRecipeAsync(long id)
                 throw new RepositoryDataDeleteException(string.Format(Resources.RecipeRepository_No_recipe_matches_with_Id_0, id));
             }
 
+            ulong authorId = GetAuthorId(entityToDelete.Author.AuthorId, entityToDelete.RecipeEntityId);
+
             context.RecipeEntities.Remove(entityToDelete);
             await context.SaveChangesAsync();
 
-            return new RecipeEntryData(entityToDelete.RecipeEntityId, entityToDelete.RecipeTitle, entityToDelete.Author.AuthorName);
+            return new RecipeEntryData(entityToDelete.RecipeEntityId, entityToDelete.RecipeTitle, authorId);
         }
         catch (DbUpdateException ex)
         {
-            throw new RepositoryDataSaveException(ex.Message, ex);
+            throw new RepositoryDataDeleteException(ex.Message, ex);
+        }
+    }
+
+    /// <summary>
+    /// Gets the author id based on its input arguments.
+    /// </summary>
+    /// <param name="authorId">The author id to parse.</param>
+    /// <param name="recipeId">The id of the recipe the author belongs to.</param>
+    /// <returns>An <see cref="ulong"/> representing the author id.</returns>
+    /// <exception cref="RepositoryDataDeleteException">Thrown when the author id could not be retrieved successfully.</exception>
+    private static ulong GetAuthorId(string authorId, long recipeId)
```

Comment:
Method name seems to be a bit weird as there's  also a  RecipeDataReader.Read procedure. Implementation seems to be out of sync / not consistent